### PR TITLE
Port `task` instance types to `runner`

### PR DESCRIPTION
**Pull request by 0x2b3bfa0**
Closes https://github.com/iterative/cml/issues/1361

### CODE REVIEW

File: iterative/kubernetes/provider.go [hunk 1]
```diff
@@ -408,6 +413,23 @@ func getInstanceType(instanceType string, instanceGPU string) (map[string]map[st
 		}, nil
 	}
 
+	
+	if match := regexp.MustCompile(`^(\d+)-(\d+)(?:\+([^*]+)\*([1-9]\d*))?$`).FindStringSubmatch(size); match != nil {
```
Comment (dacbd) [inline iterative/kubernetes/provider.go:417]: this was pulled from the task code right? can we easily abstract it out so doesn't exist in two places?

Comment (0x2b3bfa0) [reply iterative/kubernetes/provider.go:417]: > _this was pulled from the task code right?_

Yes[^1]

> _can we easily abstract it out so doesn't exist in two places?_

No[^2]

[^1]: With a few minor tweaks.
[^2]: Unless you want to review a dozen pull requests.

Comment (dacbd) [reply iterative/kubernetes/provider.go:417]: or are the tweaks different enough that it wouldn't matter?

Comment (0x2b3bfa0) [reply iterative/kubernetes/provider.go:417]: Apart from the regular expression, there is nothing in common. I'd merge this as-is. 😅

**Review by dacbd — APPROVED** [review]

**Pull request merged**